packages/atlas-service/src/main.ts

```diff
@@ -559,6 +570,29 @@ export class AtlasService {
     }
   }
 
+  static async signOut(): Promise<void> {
+    // Reset and recreate event emitter first so that we don't accidentally
+    // react on any old plugin instance events
+    this.oidcPluginLogger.removeAllListeners();
+    this.oidcPluginLogger = new EventEmitter();
+    this.attachOidcPluginLoggerEvents();
+    // Destroy old plugin and setup new one
+    await this.plugin?.destroy();
+    this.setupPlugin();
+    // Revoke tokens. Revoking refresh token will also revoke associated access
+    // tokens
+    // https://developer.okta.com/docs/guides/revoke-tokens/main/#revoke-an-access-token-or-a-refresh-token
+    await this.revoke({ tokenType: 'refresh_token' });
+    // Reset service state
+    this.token = null;
+    this.oidcPluginSyncedFromLoggerState = 'unauthenticated';
+    this.oidcPluginLogger.emit('atlas-service-signed-out');
+    // Open Atlas sign out page to end the browser session created for sign in
+    void shell.openExternal(
+      'https://account.mongodb.com/account/login?signedOut=true'
+    );
```

Comment:
Does this automatically log out users out of their Atlas session in the browser? I think I’d be surprised as a user, I would expect “Disconnect” to really only log me out inside of Compass but not Atlas in general

Reply:
This is what every other desktop app with oauth does and when discussed with cloud security they said that this is what we should do too

Reply:
Alright – surprising, but well, as Bernie mentioned, probably not too common a use case anyway :)